python/triton/compiler/compiler.py

```diff
@@ -106,28 +90,38 @@ def parse_options(self):
 
 class IRSource:
 
-    def __init__(self, path):
+    def __init__(self, path, context):
```

Comment:
Why are we passing in the context instead of creating it inside the constructor?